stable_diffusion_jax/modeling_unet2d.py

```diff
@@ -725,29 +732,40 @@ def __call__(self, sample, timesteps, encoder_hidden_states, deterministic=True)
         return sample
 
 
-class UNet2DPretrainedModel(FlaxPreTrainedModel):
-    config_class = UNet2DConfig
+class UNet2D(ModelMixin, ConfigMixin):
```

Comment:
Hmm not sure whether we should follow Transformers design here - maybe it's actually easier to directly work with `flax.linen.Module's` only. E.g.:

```python
class UNet2D(nn.Module, ModelMixin, ConfigMixin):

def __init__(...)

def setup(...)

def __call__(...)

```